### PR TITLE
fix: remediate batch 2 clang-tidy violations

### PR DESCRIPTION
**Pull request by vtz**
## Summary

Second batch of clang-tidy remediations targeting performance, readability, and code organization violations across 15 source files.

### Changes by category

**Performance** (17 fixes)
- `performance-unnecessary-value-param`: Use `std::move` for callback parameters instead of copying (event_subscriber, rpc_client, rpc_server)
- `performance-avoid-endl`: Replace `std::endl` with `'\n'` (sd_message, tp_reassembler)
- Pass `MessagePtr` by `const&` in rpc_server private helpers where only read access is needed

**Readability** (28 fixes)
- `readability-static-accessed-through-instance`: Access `E2EHeader::get_header_size()` via class name (message.h, message.cpp)
- `readability-simplify-boolean-expr`: Simplify `has_valid_method_id()` and `has_valid_length()` returns
- `readability-redundant-member-init`: Remove redundant `{}` on `UdpConfig::multicast_interface`
- `readability-qualified-auto`: Use `const auto*` for pointer-to-const (sd_client, sd_server)
- `cppcoreguidelines-prefer-member-initializer`: Move `TpTransfer::last_activity` to initializer list

**Misc / Organization** (17 fixes)
- `misc-use-anonymous-namespace`: Move file-local statics to anonymous namespaces (e2e_crc, freertos/memory, threadx/memory, sd_client, sd_server)
- `cppcoreguidelines-pro-type-cstyle-cast`: Replace C-style casts with `reinterpret_cast` in socket code (tcp_transport)

### Tests added
- Boundary tests for `has_valid_method_id()` (0x0000, 0x0001, 0xFFFE, 0xFFFF)
- Boundary tests for `has_valid_length()` (0, 7, 8, 9, UINT32_MAX)
- `E2EHeader::get_header_size()` static access verification
- `TpTransfer` timestamp initialization consistency test

## Test plan
- [x] All existing tests pass (Host, FreeRTOS, ThreadX, Zephyr)
- [x] New boundary tests pass
- [x] Clang-tidy quality gate passes
- [x] No regressions in cross-compilation (ARM Cortex-M4)

Ref: #222

Made with [Cursor](https://cursor.com)

## Summary by CodeRabbit

* **Refactor**
  * Internal cleanup and encapsulation across messaging, transport, and platform layers; reduced copies by forwarding ownership of callbacks and handlers.
  * CRC table and initialization made lazy/static-initialized for simpler startup.

* **Stability**
  * Safer initialization, tighter const-correctness, and stricter message-pool validation to reduce edge-case errors.
  * Replaced flush-causing log endings and improved socket/transport cast safety.

* **Tests**
  * Added tests for message validation boundaries, E2E header size access, move semantics, and transfer timestamp initialization.

### CODE REVIEW

**Comment by vtz** [comment]
### Review comment triage

**Fixed in next push:**
1. **FreeRTOS pool ownership** (Major) — Added bounds + alignment validation before destroying the Message
2. **CRC32 thread-safety** (Critical) — Replaced mutable globals with thread-safe `static const` local initialization via C++ magic statics
3. **`sub_info` std::move** (Trivial) — Now moved into the map instead of copied
4. **`interface_version_` move assignment** (Major) — Fixed to preserve source value instead of resetting to constant
5. **No-op total-size comparison** (Trivial) — Removed redundant identical comparison, kept the real `length_` check
6. **`test_tp.cpp` total_size** (Critical) — Already fixed in 05c208b

**Pre-existing, out of scope for this PR:**
- **`field_requests_` key mismatch** (`event_subscriber.cpp:170-188`) — The `make_field_key(service_id, instance_id, event_id)` vs `make_field_key(service_id, 0, event_id)` inconsistency and the missing cleanup on `send_message` failure are pre-existing issues not introduced by this PR. These should be tracked and fixed separately.

File: src/platform/freertos/memory.cpp [hunk 1]
```diff
@@ -57,6 +59,31 @@ static void ensure_pool_init() {
...
+    auto* raw = reinterpret_cast<char*>(msg);
+    auto* pool_start = static_cast<char*>(pool_buffer);
+    if (raw < pool_start || raw >= pool_start + POOL_SIZE * sizeof(someip::Message)) {
+        return;
+    }
+    size_t const byte_offset = static_cast<size_t>(raw - pool_start);
+    if (byte_offset % sizeof(someip::Message) != 0) {
+        return;
+    }
+    size_t const index = byte_offset / sizeof(someip::Message);
```
[the remark this answers is not in the document]
Comment (vtz) [reply src/platform/freertos/memory.cpp:76]: will be fixed with the next batch.
